.travis.yml

```diff
@@ -17,25 +17,42 @@ env:
 cache:
   directories:
     - "$CACHE_ROOT_DIR"
+    - vendor
 
-install:
-  - source bin/travis_scripts/setup_tools_dir.sh $BUILD_TOOLS_PATH
-  - source bin/travis_scripts/ensure_glide.sh $BUILD_TOOLS_PATH "v0.13.1"
-  - glide "-home" $GLIDE_HOME install
+install: skip
+
+script: echo "I haz to be overriden!"
+
+jobs:
+  include:
+    #deps cache stage
+    - stage: dep-cache
+      script:
+        - rm -rf vendor
+        - source bin/travis_scripts/setup_tools_dir.sh $BUILD_TOOLS_PATH
+        - source bin/travis_scripts/ensure_glide.sh $BUILD_TOOLS_PATH "v0.13.1"
+        - glide "-home" $GLIDE_HOME install
+    #test stage
+    - stage: test
+      script: bin/check_vet
+    - script: bin/builder_run bin/check_lint
+    - script: bin/check_fmt
+    - script: bin/check_license
+    - script: bin/test
+    #e2e stage
+    - stage: e2e-test
+      script: bin/travis_scripts/check_docker.sh && bin/travis_scripts/check_docker_compose.sh && bin/test_e2e
```

Comment:
Not much value of those checks, drop them maybe?
Could they've be in some `before_*` phase?